speechbrain/k2_integration/graph_compiler.py

```diff
@@ -0,0 +1,482 @@
+
+    def decode(self,
+               log_probs: torch.Tensor,
+               input_lens: torch.Tensor,
+               search_beam=5,
+               output_beam=5,
+               ac_scale=1.0,
+               min_active_states=300,
+               max_active_states=1000,
+               is_test: bool = True,
+               decoding_method: str = "1best",
+               lm_scale_list: Optional[List[float]] = None,
+               rescoring_lm_path: Optional[Path] = None
+        ) -> Union[List[str], Dict[str, List[str]]]:
+        """
+        Decode the given log_probs with self.decoding_graph without language model.
+
+        Arguments
+        ---------
+        log_probs: torch.Tensor
+            It is an input tensor of shape (batch, seq_len, num_tokens).
+        input_lens: torch.Tensor
+            It is an int tensor of shape (batch,). It contains lengths of
+            each sequence in `log_probs`.
+        search_beam: int
+            decoding beam size
+        output_beam: int
+            lattice beam size
+        ac_scale: float
+            acoustic scale applied to `log_probs`
+        min_active_states: int
+            minimum #states that are not pruned during decoding
+        max_active_states: int
+            maximum #active states that are kept during decoding
+        is_test: bool
+            if testing is performed then we won't log warning about <UNK>s.
+        decoding_method: str
+            one of 1best, whole-lattice-rescoring, or nbest.
+        lm_scale_list: List[float]
+            a list of language model scale factors. Defaults to [0.6].
+        rescoring_lm_path: Path
+            path to the LM to be used for rescoring. If not provided
+            and the decoding method is whole-lattice-rescoring, then you need to provide
+            the `rescoring_lm_path` in the constructor of this class.          
+
+        Returns
+        -------
+        If decoding_method==1best: a list of strings, each of which is the decoding 
+        result of the corresponding utterance.
+
+        If decoding_method==whole-lattice-rescoring: a dict of lists of strings, each of
+        which is the decoding result of the corresponding utterance. The keys are the 
+        language model scale factors used for rescoring.
+        """
+        lm_scale_list = lm_scale_list or [0.6]
+        device = log_probs.device
+        if self.decoding_graph is None:
+            if is_test:
+                self.lexicon.log_unknown_warning = False
+            if self.G_path is None:
+                self.compile_HL()
+            else:
+                logger.info("Compiling HLG instead of HL")
+                self.compile_HLG()
+            if self.decoding_graph.device != device:
+                self.decoding_graph = self.decoding_graph.to(device)
+            if decoding_method == "whole-lattice-rescoring":
+                self.rescoring_graph = self.get_rescoring_LM(str(rescoring_lm_path))\
```

Comment:
`self.get_rescoring_LM(str(rescoring_lm_path))` -> `self.get_rescoring_LM(rescoring_lm_path)` otherwise
the first line of the function get_rescoring_L
```
path = str(path or self.rescoring_lm_path)
```
gets evaluated to path = "None" instead of path = None

Reply:
Thank you @pchampio! I'll fix that.